Implement voronoi-service.ts

CubePresser opened the pull request:
Added:
 - voronoi-service.ts
 - 'lodash' to package.json
 Modified:
 - Model.ts, changed Node to Cell, simplified types

2-voronoi-pattern

CubePresser committed:
WIP Implement voronoi-service.ts

Added:
 - voronoi-service.ts
 - 'lodash' to package.json
 Modified:
 - Model.ts, changed Node to Cell, simplified types

2-voronoi-pattern

Manason commented on src/util/model.ts:
```diff
@@ -7,13 +7,11 @@ export type Point = Readonly<{
+  color: string | null;
   polygons: Polygon[];
-  neighbors: Dictionary<Node>;
+  neighbors: string[];
```
make this is a set

Manason commented on src/util/voronoi-service.ts:
```diff
@@ -0,0 +1,58 @@
+  }, {});
+
+  return cellsById;
+};
```
consider adding one line comments above each function to explain what the function does

Manason replied:
Functions are easy to understand and nicely extracted.

Manason commented on src/util/voronoi-service.ts:
```diff
@@ -0,0 +1,58 @@
+  const iterable = voronoi.neighbors(index);
+  for (const entry of iterable) {
+    // TODO: Use hash function to generate better IDs...
+    neighbors.push(`node-${entry}`);
```
cell instead of node

Manason commented on src/views/Sandbox.tsx:
```diff
@@ -18,19 +19,13 @@ const Sandbox: React.FC<SandboxProps> = ({
-    console.log('Voronoi results', voronoi);
-  }, [ voronoi ]);
+    console.log('Voronoi results', nodes);
+  }, [ nodes ]);
```
cells

Manason commented on src/views/Sandbox.tsx:
```diff
@@ -18,19 +19,13 @@ const Sandbox: React.FC<SandboxProps> = ({
-
-    return result;
+  const nodes = useMemo(() => {
+    return generateNodes([ [ 0, 0 ], [ 0, 1 ], [ 1, 0 ], [ 1, 1 ] ], [ 0, 0, 960, 500 ]);
```
cells

Manason commented on src/util/voronoi-service.ts:
```diff
@@ -0,0 +1,58 @@
+  };
+};
+
+export const generateNodes = (points: D3Point[], viewBounds: [number, number, number, number]): Dictionary<Cell> => {
```
generateCells

CubePresser committed:
Generate points + respond to PR feedback

Added:
- Psuedo Random Number Generator
- Random number util
- Generate points exposed in voronoi service

2-voronoi-pattern

CubePresser committed:
Replace instance of "nodes"

2-voronoi-pattern

CubePresser committed:
Merge remote-tracking branch 'origin/main' into 2-voronoi-pattern
